### PR TITLE
Reset de senha - Adiciona tela de criação de nova senha

### LINKED ISSUE #1047: Recuperação de Senha: Tela de criação de senha
Épico: https://github.com/pagarme/roadmap/issues/555

### Contexto

Precisamos trazer o processo de Recuperação de Senha para dentro da pilot. 

### Exemplo

Fluxo de interface: https://app.zeplin.io/project/59e66a0aa633e7a9cf5c0cf4/dashboard?q=senha

### Descrição técnica

Continuação do fluxo iniciado em issue #1045 

Quando o link de recuperação de senha recebido em email é clicado, o usuário precisa ser enviado para a nova dashboard, que interpretará o token no link e permitirá a criação da nova senha.

### Requisitos e Critérios de Aceite

- [ ] Interpretação de tokens de link 
- [ ] Tela de criação de nova senha
- [ ] Requisição para API de troca de senha

### PR DESCRIPTION
**Pull request by vagnervst**
## Contexto
Para que seja possível concluir o fluxo de redefinição de senha via Pilot, precisamos da tela de criação da nova senha. Este PR adiciona esta tela.

## Checklist
- [x] Não permite acessar a página sem um `token`

## Issues linkadas
- [x] Resolves #1047

### Preview:

image

## Como testar?
1. Gere um token de reset de senha que será enviado no e-mail:
    ```
    curl -X PUT \
      https://api.pagar.me/1/users/reset_password \
      -H 'Content-Type: application/json' \
      -d '{
  	    "email": "meu@email.com"
      }'
    ```

2. Pegue este token da URL da página de reset de senha da Dash atual:
  image

3. Acesse a pilot utilizando o mesmo token, na URL `http://localhost:3000/#/account/password/reset/<token>`

### CODE REVIEW

**Comment by vagnervst** [comment]
@Deivis Boa! Se tratava de um setState em `.finally` que estava usando. Corrigido :slightly_smiling_face:

**Comment by Deivis** [comment]
O espaçamento entre o logo e o texto `DIGITE SUA NOVA SENHA PARA CONCLUIR A REDEFINIÇÃO.` está muito grande.
image

File: packages/pilot/public/locales/pt/translations.json [hunk 1]
```diff
@@ -755,6 +750,18 @@
...
       "wrong_pass": "Senha inválida"
     },
+    "password_recovery": {
+      "confirmation": "Em instantes você receberá um e-mail com as instruções para a redefinição de sua senha.",
+      "confirmation_emphasis": "Pronto!",
+      "recovery_call": "Digite o e-mail cadastrado para recuperar sua senha.",
+      "recovery_action": "Enviar"
+    },
+    "password_reset": {
+      "confirmation": "Sua nova senha foi criada com sucesso. Retorne para o Login para começar a utiliza-la.",
```
Comment (gmbarroso) [inline packages/pilot/public/locales/pt/translations.json]: Sugestão: "... Retorne para a área de login ..."

Comment (vagnervst) [reply packages/pilot/public/locales/pt/translations.json]: Boa! Subi um fix com a sugestão :slightly_smiling_face:

File: packages/pilot/public/locales/pt/translations.json [hunk 2]
```diff
@@ -755,6 +750,22 @@
...
       "wrong_pass": "Senha inválida"
     },
+    "password_recovery": {
+      "confirmation": "Em instantes você receberá um e-mail com as instruções para a redefinição de sua senha.",
+      "confirmation_emphasis": "Pronto!",
+      "recovery_call": "Digite o e-mail cadastrado para recuperar sua senha.",
+      "recovery_action": "Enviar"
+    },
+    "password_reset": {
+      "confirmation": "Sua nova senha foi criada com sucesso. Retorne para a área de Login para começar a utiliza-la.",
```
Comment (miguelssrs) [inline packages/pilot/public/locales/pt/translations.json]:
```suggestion
      "confirmation": "Sua nova senha foi criada com sucesso! Retorne para a área de Login para começar a utilizá-la.",
```

Comment (vagnervst) [reply packages/pilot/public/locales/pt/translations.json]: Feito!

**Comment by vagnervst** [comment]
@Deivis Adicionei num commit separado essa validação, fazendo o botão aparecer somente na tela de Login. Porém tive que validar pelo `pathname` pois o component `Presentation` é utilizado fora dos containers de cadastro/reset de senha :slightly_frowning_face:

**Comment by rsmelo** [comment]
as stories de  `password` na sessão `login` estão precisando de alguns refactors que estão dando erro no console
`
Warning: Failed prop type: The prop `onBackToLogin` is marked as required in `ResetConfirmation`, but its value is `undefined`.
    in ResetConfirmation (at Landing/index.js:142) checkPropTypes.js:20
    printWarning checkPropTypes.js:20
    checkPropTypes checkPropTypes.js:82
    React 2
    js index.js:141
    defaultDecorateStory client_api.js:72
    checkA11y index.js:61
    defaultDecorateStory client_api.js:71
    defaultDecorateStory client_api.js:72
    ThemeDecorator themeDecorator.js:8
    defaultDecorateStory client_api.js:71
    defaultDecorateStory client_api.js:72
    withSubscriptionTracking client_api.js:95
    defaultDecorateStory client_api.js:71
    getStoryWithContext story_store.js:196
    renderMain render.js:61
    renderMain start.js:208
    renderUI start.js:228
    dispatch Redux
    _renderMain config_api.js:98
    render config_api.js:42
    ConfigApi config_api.js:71
    js percyIntegration.js:17
    js main.f05095ceeb780fc90440.bundle.js:20722
    Webpack 10
Warning: Failed prop type: The prop `onBackToLogin` is marked as required in `Confirmation`, but its value is `undefined`.
    in Confirmation (at Confirmation.js:10)
    in ResetConfirmation (at Landing/index.js:142)
    in div (at Account/index.js:31)
    in div (at Account/index.js:30)
    in div (created by Col)
    in Col (created by ThemedCol)
    in ThemedCol (created by LandingPrimarySection)
    in LandingPrimarySection (created by ThemedLandingPrimarySection)
    in ThemedLandingPrimarySection (at Account/index.js:29)
    in div (created by Row)
    in Row (created by ThemedRow)
    in ThemedRow (created by Landing)
    in div (created by Grid)
    in Grid (created by ThemedGrid)
    in ThemedGrid (created by Landing)
    in div (created by Landing)
    in Landing (created by ThemedLanding)
    in ThemedLanding (at Account/index.js:28)
    in Account (at Landing/index.js:139)
    in ThemeProvider (at themeDecorator.js:8) checkPropTypes.js:20
`setOptions(options)` is deprecated. Please use the `withOptions(options)` decorator globally. browser.js:39
[HMR] connected client.js:95
Warning: Failed prop type: The prop `onChange` is marked as required in `PasswordResetForm`, but its value is `undefined`.
    in PasswordResetForm (at Landing/index.js:128) checkPropTypes.js:20
Warning: Failed prop type: The prop `validations` is marked as required in `PasswordResetForm`, but its value is `undefined`.
    in PasswordResetForm (at Landing/index.js:128) checkPropTypes.js:20
    printWarning checkPropTypes.js:20
    checkPropTypes checkPropTypes.js:82
    React 2
    js index.js:127
    defaultDecorateStory client_api.js:72
    checkA11y index.js:61
    defaultDecorateStory client_api.js:71
    defaultDecorateStory client_api.js:72
    ThemeDecorator themeDecorator.js:8
    defaultDecorateStory client_api.js:71
    defaultDecorateStory client_api.js:72
    withSubscriptionTracking client_api.js:95
    defaultDecorateStory client_api.js:71
    getStoryWithContext story_store.js:196
    renderMain render.js:61
    renderMain start.js:208
    renderUI start.js:228
    dispatch Redux
    start start.js:144
    _handleEvent index.js:193
    _handleEvent index.js:192
    Channel index.js:68
    _handleEvent index.js:128
Warning: Failed prop type: The prop `base` is marked as required in `PasswordRecoveryForm`, but its value is `undefined`.
    in PasswordRecoveryForm (at Landing/index.js:98) checkPropTypes.js:20`

**Comment by gmbarroso** [comment]
Uma sugestão:
image

Em tela inteira, o texto fica muito colado com o link do GitHub. Não precisa modificar agora, mas seria legal colocar um espaço entre os dois.

**Comment by vagnervst** [comment]
Cancelei um dos approves pois encontrei um fix que subirei em alguns minutos

**Comment by vagnervst** [comment]
@gmbarroso @rsmelo Testando com o core local notei que na request de solicitação de reset de senha, o pagarme-js envia os parâmetros de e-mail via URL, logo havia a necessidade de fazer o encoding desse e-mail (para mantermos o `+`):

https://github.com/pagarme/pilot/pull/1357/files?file-filters%5B%5D=.js#diff-49b93c10f40ee02ec62fd85317e1722eR43

Fora esta mudança, adicionei um espaçamento à direita no texto de copyright, como reportado pelo @gmbarroso

**Comment by rsmelo** [comment]
@vagnervst aqueles erros da prop base no storybook voltaram a ocorrer nas stories de Login e Singup
`
Warning: Failed prop type: The prop `base` is marked as required in `PasswordRecoveryForm`, but its value is `undefined`.
    in PasswordRecoveryForm (at Landing/index.js:98) checkPropTypes.js:20
Warning: Failed prop type: The prop `onChange` is marked as required in `PasswordResetForm`, but its value is `undefined`.
    in PasswordResetForm (at Landing/index.js:128) checkPropTypes.js:20
Warning: Failed prop type: The prop `onBackToLogin` is marked as required in `ResetConfirmation`, but its value is `undefined`.
    in ResetConfirmation (at Landing/index.js:142) checkPropTypes.js:20
Warning: Failed prop type: The prop `onBackToLogin` is marked as required in `Confirmation`, but its value is `undefined`.
    in Confirmation (at Confirmation.js:10)
    in ResetConfirmation (at Landing/index.js:142)
    in div (at Account/index.js:31)
    in div (at Account/index.js:30)
    in div (created by Col)
    in Col (created by ThemedCol)
    in ThemedCol (created by LandingPrimarySection)
    in LandingPrimarySection (created by ThemedLandingPrimarySection)
    in ThemedLandingPrimarySection (at Account/index.js:29)
    in div (created by Row)
    in Row (created by ThemedRow)
    in ThemedRow (created by Landing)
    in div (created by Grid)
    in Grid (created by ThemedGrid)
    in ThemedGrid (created by Landing)
    in div (created by Landing)
    in Landing (created by ThemedLanding)
    in ThemedLanding (at Account/index.js:28)
    in Account (at Landing/index.js:139)
    in ThemeProvider (at themeDecorator.js:8) checkPropTypes.js:20
Warning: Failed prop type: The prop `base` is marked as required in `SignUpForm`, but its value is `undefined`.
    in SignUpForm (at Landing/index.js:160)`

**Comment by vagnervst** [comment]
@rsmelo boa! Subi as correções novamente

**Comment by rsmelo** [comment]
@vagnervst percebi um problema que na verdade é problema do pagarme-js. Os dados (password e token) estão sendo enviados como query string e acaba quebrando quando a senha tem caracteres especiais e dai a senha que eu ponho não é de fato que acaba sendo alterada

**Comment by Deivis** [comment]
Faltou a valicação de força de senha como foi feito no "Cadastro pós convite"
image

**Comment by vagnervst** [comment]
@Deivis Corrigi os pontos citados!

**Comment by rsmelo** [comment]
@vagnervst lançada versão do pagarme-js com o fix do `PUT` https://github.com/pagarme/pagarme-js/releases/tag/4.4.1

**Comment by rsmelo** [comment]
na tela de informar o e-mail o container está posicionado muito acima, na tela de mensagem de sucesso tambem

### prints
image
image

### zeplin
image
image


link do zeplin https://app.zeplin.io/project/59e66a0aa633e7a9cf5c0cf4/screen/5a677466afb8e9b2e167879f

**Comment by vagnervst** [comment]
@rsmelo No commit https://github.com/pagarme/pilot/pull/1357/commits/8e6891893d1ed76d3742e684bd0ffaf23606eddb refatorei a estilização destes componentes de Account, acredito que melhorou agora.

**Comment by rsmelo** [comment]
@vagnervst 
nessa tela o espaçamento entre a mensage de "digite o email cadastrado" e o input estão com bastante espaçamento se comparado ao zeplin.
Na segunda tela está um pouco mais parecido mas ainda tem um pouco a mais de espaçamento

image
image

**Comment by Deivis** [comment]
Screen Shot 2019-09-18 at 10 24 36
Faltou aquele `Voltar para o login` maroto :)

**Comment by vagnervst** [comment]
@rsmelo @Deivis Feito!

File: packages/pilot/src/containers/Account/style.css [hunk 3]
```diff
@@ -87,33 +88,42 @@
...
 
-  & .recoveryCall {
-    text-transform: uppercase;
+  &.passwordReset {
+    min-height: 300px;
+  }
+}
+
+.confirmationContent {
+  margin-top: 131px;
```
Comment (Deivis) [inline packages/pilot/src/containers/Account/style.css]: E este número mágico de `131px` ?

Comment (vagnervst) [reply packages/pilot/src/containers/Account/style.css]: É a margem exata que está no zeplin, sugere utilizar uma das variáveis de `spacing`? Pq mudaria bastante do que temos lá

**Pull request merged**